rust-module/mfrc522.rs

```diff
@@ -104,10 +103,11 @@ impl KernelModule for Mfrc522Driver {
             miscdev::Registration::new_pinned::<Mfrc522FileOps>(cstr!("mfrc522_chrdev"), None, ())?;
 
         let mut spi =
-            spi::DriverRegistration::new(&THIS_MODULE, cstr!("mfrc522")).with_probe(mfrc522_probe);
-        spi.register()?;
+            spi::DriverRegistration::new_pinned(&THIS_MODULE, cstr!("mfrc522"))?
+            .with_probe(mfrc522_probe);
```

Comment:
can't we just call `.register` here ?

Reply:
Yeah, that's why the FIXME is here for now cause I think it's ugly to do `spi.as_mut().register()`.

We have a solution, which helps with cleanliness: remove the builder pattern (which I agree with) and pass all spi methods in the `new_pinned` function, and register the driver in the `new_pinned` function (this is what miscdev.rs does). The downside is that the function takes 3 more arguments, which will be `None` most of the time

Reply:
I'm removing the builder pattern